src/Tizen.Sensor/Tizen.Sensor/Sensor.cs

```diff
@@ -541,9 +689,778 @@ private bool CheckSensorHandle()
+            {
+                Log.Error(Globals.LogTag, "Recorder query is not created");
+            }
+        }
+
+        /// <summary>
+        ///  Sets a time-type query parameter.
+        /// </summary>
+        /// <param name="recorderQueryE">
+        /// Query parameter.
+        /// </param>
+        /// <param name="time">
+        /// Time.
+        /// </param>
+        /// <since_tizen> 6 </since_tizen>
+        public void RecorderQuerySetTime(int recorderQueryE, int time)
+        {
+            if (CheckRecorderQuery()) { 
+                int error = Interop.SensorRecoder.RecorderQuerySetTime(recorderQuery, recorderQueryE, time);
+                if (error != (int)SensorError.None)
+                {
+                    Log.Error(Globals.LogTag, "Error checking sensor recorder query set time");
+                    throw SensorErrorFactory.CheckAndThrowException(error, "Sensor.RecorderQuerySetTime Failed");
+                }
+            }
+            else
+            {
+                Log.Error(Globals.LogTag, "Recorder query is not created");
+            }
+        }
+
+        /// <summary>
+        ///  Queries the recorded data synchronously.
+        /// </summary>
+        /// <param name="type">
+        /// Sensor type.
+        /// </param>
+        /// <since_tizen> 6 </since_tizen>
+        public void RecorderReadSync(int type)
+        {
+            if (CheckRecorderQuery())
+            {
+                _sensorRecorderDataCallBack = (int _type, int data, int remains, int _error, Int64 _userData) =>
+                {
+                    RecorderDataEventArgs e = new RecorderDataEventArgs(_type, data, remains, _error, _userData)
+                    {
+                    };
+
+                    if (_error != 0)
+                    {
+                        Log.Error(Globals.LogTag, "Error in Sensor Recorder Data Read Synchronously CallBack");
+                    }
+
+                    RecorderDataReceived?.Invoke(null, e);
+                    Log.Debug(Globals.LogTag, "Recorder data callback recieved");
+                    return true;
+                };
+
+                int error = Interop.SensorRecoder.RecorderReadSync(type, recorderQuery, _sensorRecorderDataCallBack, IntPtr.Zero);
+                if (error != (int)SensorError.None)
+                {
+                    Log.Error(Globals.LogTag, "Error in synchronously reading recorder");
+                    throw SensorErrorFactory.CheckAndThrowException(error, "Sensor.RecorderRead Failed");
+                }
+            }
+            else
+            {
+                Log.Error(Globals.LogTag, "Recorder query is not created");
+            }
+        }
+
+        /// <summary>
+        ///  Queries the recorded data asynchronously.
+        /// </summary>
+        /// <param name="type">
+        /// Sensor type.
+        /// </param>
+        /// <since_tizen> 6 </since_tizen>
+        public void RecorderRead(int type)
```

Comment:
You need to study about aync/await
```suggestion
        public Task<int> RecorderReadAsync(int type)
```

Reply:
added async API using await keyword.